File: python/pe020.py
#!/usr/bin/env python3

""" this computes projecteuler.net problem 020 """

import sys

LAST = 100
factorial = 1
for i in range(LAST):
    factorial *= i + 1

fstring = str(factorial)
flen = len(fstring)

fsum = 0
for i in range(flen):
    fsum += int(fstring[i])

print(fsum)
sys.exit(0)

File: ruby/pe020.rb
#!/usr/bin/ruby

last = 100
factorial = 1
(1 .. last).each do |i|
  factorial *= i
end

fsum = 0
fstring = factorial.to_s
fstring.split('').each do |i|
  fsum += i.to_i
end

puts(fsum)
